subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientHelper.java

```diff
@@ -155,6 +160,11 @@ private CloseableHttpResponse processResponse(String source, String method, Clos
         }
         if (!wasSuccessful(response)) {
             LOGGER.info("Failed to get resource: {}. [HTTP {}: {}]", method, response.getStatusLine(), source);
+
+            if (wasUnrecoverable5xx(response)) {
```

Comment:
❌ Why introduce a new type? We already have one for http errors and we can add something like `isSeverError`

Reply:
done